pallets/core/src/common/authorization.rs

```diff
@@ -26,18 +37,25 @@ where
     <Self::Signer as Deref>::Target: AuthorizeTarget<A::Target, Self::Key>,
 {
     /// This signature allows `Self::Signer` to perform the supplied action.
-    fn authorizes_signed_action<T: did::Config>(&self, action: &A) -> AuthorizationResult<T, Self>
+    fn authorizes_signed_action<T: did::Config>(
+        &self,
+        action: &A,
+        value: Option<&<A::Target as Associated<T>>::Value>,
+    ) -> AuthorizationResult<Self>
     where
         A: ToStateChange<T>,
+        A::Target: Associated<T>,
     {
         let signer_pubkey = self.key::<T>().ok_or(did::Error::<T>::NoKeyForDid)?;
         let encoded_state_change = action.to_state_change().encode();
 
         let signer = self.signer().ok_or(did::Error::<T>::InvalidSigner)?;
-        (*signer).ensure_authorizes_target(&signer_pubkey, action)?;
-        signer.ensure_authorizes_target(&signer_pubkey, action)?;
+        (*signer).ensure_authorizes_target(&signer_pubkey, action, value.as_ref().copied())?;
```

Comment:
Why is `ensure_authorizes_target` called twice on the reference and value? Also can you document the type constraints on `AuthorizeSignedAction`?

Reply:
Because it will ensure that both the wrapper type (for ex. `Convener`) and underlying type (`Did` or `DidMethodKey`) authorize this target.
On the wrapper level we usually check specific rules like owning a registry or being a `Controller` for a `Did` while on the `Did`/`DidMethodKey` level we check that provided keys are valid.